core/dbt/lib.py

```diff
@@ -13,6 +17,46 @@ class RuntimeArgs:
     target: str
 
 
+class SqlCompileRunnerNoIntrospection(SqlCompileRunner):
+    def compile_and_execute(self, manifest, ctx):
```

Comment:
Looks like this function is the same as `compile_and_execute` [here](https://github.com/dbt-labs/dbt-core/blob/7b1d61c956d47be759faebf2f479334e585c7585/core/dbt/task/base.py#L313) with line 313 `with self.adapter.connection_for(self.node):` removed. Looks good now but I would recommend add a TODO to refactor here when doing runtime(maybe?) so we are not duplicating majority of this function in two places.